Mathlib/AlgebraicGeometry/Restrict.lean

```diff
@@ -515,6 +515,68 @@ instance {X Y : Scheme.{u}} (f : X ⟶ Y) (U : Y.Opens) [IsOpenImmersion f] :
   delta morphismRestrict
   exact PresheafedSpace.IsOpenImmersion.comp _ _
 
+variable {X Y : Scheme.{u}}
+
+namespace Scheme.Hom
+
+/-- The restriction of a morphism `f : X ⟶ Y` to open sets on the source and target. -/
+def resLE (f : Hom X Y) (U : Y.Opens) (V : X.Opens) (e : V ≤ f ⁻¹ᵁ U) : V.toScheme ⟶ U.toScheme :=
+  (X.restrictFunctor.map (homOfLE e)).left ≫ f ∣_ U
+
+variable (f : X ⟶ Y) {U U' : Y.Opens} {V V' : X.Opens} (e : V ≤ f ⁻¹ᵁ U)
+
+lemma resLE_eq_morphismRestrict : f.resLE U (f ⁻¹ᵁ U) le_rfl = f ∣_ U := by
+  simp [Scheme.Hom.resLE]
+
+@[simp]
+lemma resLE_comp_ι : f.resLE U V e ≫ U.ι = V.ι ≫ f := by
+  simp [resLE, restrictFunctor_map_ofRestrict_assoc]
+
+@[reassoc (attr := simp)]
+lemma map_resLE (i : V' ⟶ V) :
+    (X.restrictFunctor.map i).left ≫ f.resLE U V e = f.resLE U V' (i.le.trans e) := by
+  simp only [Scheme.Hom.resLE, homOfLE_leOfHom, ← Over.comp_left_assoc, ← Functor.map_comp]
+  rfl
+
+/-- Variant of `map_resLE` for equality. -/
+@[reassoc]
+lemma map_resLE' (i : V' = V) :
+    (X.restrictFunctor.map (eqToHom i)).left ≫ f.resLE U V e = f.resLE U V' (i ▸ e) :=
+  map_resLE _ _ _
```

Comment:
Do we need this?

Reply:
I don't know, I just copied it from the `appLE` case. Removed it for now.